src/Gaffer/VTuneMonitor.cpp

```diff
@@ -0,0 +1,126 @@
+//////////////////////////////////////////////////////////////////////////
+//
+//  Copyright (c) 2016, Image Engine Design Inc. All rights reserved.
+//
+//  Redistribution and use in source and binary forms, with or without
+//  modification, are permitted provided that the following conditions are
+//  met:
+//
+//      * Redistributions of source code must retain the above
+//        copyright notice, this list of conditions and the following
+//        disclaimer.
+//
+//      * Redistributions in binary form must reproduce the above
+//        copyright notice, this list of conditions and the following
+//        disclaimer in the documentation and/or other materials provided with
+//        the distribution.
+//
+//      * Neither the name of Image Engine Design Inc nor the names of
+//        any other contributors to this software may be used to endorse or
+//        promote products derived from this software without specific prior
+//        written permission.
+//
+//  THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS
+//  IS" AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO,
+//  THE IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR
+//  PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT OWNER OR
+//  CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL,
+//  EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO,
+//  PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR
+//  PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF
+//  LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING
+//  NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE OF THIS
+//  SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+//
+//////////////////////////////////////////////////////////////////////////
+
+#include "Gaffer/VTuneMonitor.h"
+
+#ifdef GAFFER_VTUNE
+
+#include <stdio.h>
+
+#include "ittnotify.h"
+#include "Gaffer/Process.h"
+#include "Gaffer/Plug.h"
+#include "Gaffer/Node.h"
+
+struct OpaqueDomain
+{
+	OpaqueDomain(const std::string& domain) : imp ( __itt_domain_create( domain.c_str() ) )
+	{}
+
+	__itt_domain *imp;
+};
+
+namespace
+{
+	const IECore::InternedString hashProcessType( "computeNode:hash" );
+
+	typedef const Gaffer::Node* TaskHandleKey;
+
+	typedef std::map<TaskHandleKey, __itt_string_handle*> TaskHandleMap;
+	struct ThreadLocalData
+	{
+		TaskHandleMap taskHandleMap;
+		bool enabled;
+	};
+
+	tbb::enumerable_thread_specific<ThreadLocalData, tbb::cache_aligned_allocator<ThreadLocalData> > g_threadData;
+}
+
+using namespace Gaffer;
+
+VTuneMonitor::VTuneMonitor(const std::string& domain /*= "Gaffer" */, bool monitorHashProcess /* = false */ )
+: m_monitorHashProcess( monitorHashProcess )
+{
+	m_domain.reset( new OpaqueDomain( domain ) );
```

Comment:
What happens when two VTuneMonitors are created with the same domain name? Are we allowed to call `__itt_domain_create` twice with the same value? How useful is the name in practice, and do you think we should follow the guidelines in the documentation and use "org.gafferhq.gaffer" as the domain? Basically, I'm wondering if we shouldn't just create a single static domain...